pkgs/development/python-modules/python-miio/default.nix

```
@@ -0,0 +1,33 @@
+{ stdenv
+, buildPythonPackage
+, fetchPypi
+, appdirs
+, click
+, construct
+, cryptography
+, zeroconf
+, attrs
+, pytz
+, tqdm
+, netifaces
+}:
+
+buildPythonPackage rec {
+  pname = "python-miio";
+  version = "0.4.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "19423b3386b23d2e0fc94a8f6a358bcfbb44eed05376e33fd434d26d168bd18c";
+  };
+
+  propagatedBuildInputs = [ appdirs click construct cryptography zeroconf attrs pytz tqdm netifaces ];
+
```

```suggestion
  checkInputs = [ pytest ];
```
```
builder for '/nix/store/47qf3h62fp3flwxjzv5dpl87hl9cl34m-python3.8-python-miio-0.4.8.drv' failed with exit code 1; last 10 log lines:
      import pytest
  ModuleNotFoundError: No module named 'pytest'


  ----------------------------------------------------------------------
  Ran 48 tests in 0.043s

  FAILED (errors=44)
  Test failed: <unittest.runner.TextTestResult run=48 errors=44 failures=0>
  error: Test failed: <unittest.runner.TextTestResult run=48 errors=44 failures=0>
```

---

Somehow I forgot to push that line. Sorry about that. With `pytest` added the test unfortunately still fail:

```
======================================================================
FAIL: test_set_scene (miio.tests.test_yeelight.TestYeelight)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-miio-0.4.8/miio/tests/test_yeelight.py", line 220, in test_set_scene
    self.fail()
AssertionError: None

----------------------------------------------------------------------
Ran 322 tests in 0.137s

FAILED (failures=17, errors=331)
Test failed: <unittest.runner.TextTestResult run=322 errors=331 failures=17>
error: Test failed: <unittest.runner.TextTestResult run=322 errors=331 failures=17>
```

---

Not sure how they are executing this through setuptools, but running pytest directly seems to fix the test suite issues.
```suggestion
  checkPhase = ''
    pytest
  '';
```

---

That looks good. Thanks for your help.